File: lib/models/recipe.dart
// lib/models/recipe.dart
class Recipe {
  final String title;
  final String category;
  final List<String> ingredients;
  final String preparation;
  //final String imageUrl;

  Recipe({
    required this.title,
    required this.category,
    required this.ingredients,
    required this.preparation,
    //required this.imageUrl,
  });
}

File: api/models/recipe.js
const mongoose = require("mongoose");

const recipeSchema = new mongoose.Schema(
  {
    category_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "category",
      required: true,
    },
    description: { type: String, required: true },
    ingredients: [{ type: mongoose.Schema.Types.ObjectId, ref: "ingredient" }],
    howtomake: { type: String, required: true },
    observations: { type: String },
  },
  { collection: "recipe" }
);

module.exports = mongoose.model("recipe", recipeSchema);
